Add explanation for UG command examples (#155)
AY2223S1-CS2103T-T11-3/tp: docs/UserGuide.md

* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`.
* Only the name is searched.
* Only full words will be matched e.g. `Han` will not match `Hans`.

> * Students matching at least one keyword will be returned (i.e. `OR` search)

Reviewer: ```suggestion
* Students with names matching at least one keyword will be returned (i.e. `OR` search)
```

* At least one of the optional fields must be provided.

Examples:

> * `grade 1 ra1/17 mt/55` add to the grades for the 1st student in your student list: 17 marks for RA1 and 55 marks for midterms. 

Reviewer: ```suggestion
* `grade 1 ra1/17 mt/55` records the grades for the 1st student in your student list: 17 marks for RA1 and 55 marks for midterms. 
```

* `d/DEADLINE` should be in the format `yyyy-MM-dd`

Examples:
* `addtask t/Prepare tutorial slides d/2022-11-01` adds a task with description `Prepare tutorial slides` and deadline of `Nov-11-2022` to your task list.

> * `addtask t/Mark missions` adds a task with description `Mark missions` with no deadline to your task list.

Reviewer: ```suggestion
* `addtask t/Mark missions` adds a task with description `Mark missions` and no deadline to your task list.
```

* `si/STUDENT_INDEX` can be specified one or more times according to the number of attending students.

Examples:

> * `addmc l/Mastery check 1 sd/2022-10-12 st/12:00 et/14:00 si/1 si/2` adds a mastery check lesson with description `Mastery check 1` on `Oct-20-2022` from `12:00` to `14:00` with 1st and 2nd student in your student list.

Reviewer 2: Minor issue with grammar, "with the 1st and 2nd student" might be better

> * `addmc sd/2022-11-13 st/23:00 ed/2022-11-14 et/00:00 si/1` adds a mastery check lesson from `Nov-13-2022 23:00` to `Nov-14-2022 00:00` with 1st student in your student list.

Reviewer 2: "with the 1st student"

* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​

Example:

> * `markstudent li/2 si/3` marks the 3rd student in the student list within the 2nd lesson in your lesson list as present.

Reviewer 2: I think this phrasing is unnecessary

The phrasing before is clear enough imo since the paragraph above already explains what STUDENT_INDEX refers to

Author: I think I will remove the `in the student list part within` part, but keep the back part for `in your lesson list` to be more consistent with previous example

* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​

Example:

> * `unmarkstudent li/2 si/3` marks the 3rd student in the student list within the 2nd lesson in your lesson list as absent.

Reviewer 2: Same as before

* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​

Example:

> * `addnote n/Get back to jeff on streams li/1 si/2` adds a note for the 2nd student within the 1st lesson in your lesson list.

Reviewer: ```suggestion
* `addnote n/Get back to jeff on streams li/1 si/2` adds a note for the 2nd student in the student list within the 1st lesson in your lesson list.
```

> * `addnote n/Get back to the class on streams li/2` adds a note for the 2nd lesson in your lesson list.

Reviewer 2: Same as before

* `NOTE_INDEX`, `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​

Example:

> * `deletenote ni/1 li/1 si/2` deletes the 1st note for the 2nd student within the 1st lesson in your lesson list.

Reviewer: Same as suggestion for `addnote`, I think the phrasing might be more consistent with the explanations for the other commands if you mention the student list.

> * `deletenote ni/2 li/1` deletes the 2nd note from the 1st lesson in your lesson list.

Reviewer 2: Same as before

* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​

Example:

> * `addpart p/100 li/1 si/2` sets participation for the 2nd student in the student list within the 1st lesson in your lesson list to be 100.

Reviewer 2: Same as before
